pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -103,7 +105,15 @@ func handleSyncOp(ctx *downloaderContext, key string,
 		// pass in the config.Name instead of 'filename' which
 		// does not contain the prefix of the relative path with '/'s
 		remoteName = config.Name
-		metricsURL = fmt.Sprintf("S3:%s/%s", dsCtx.Dpath, config.Name)
+		// remove all leading '/' if we have them in dsPath, remove duplicates of '/' and split by '/'
+		splittedPath := strings.Split(path.Clean(strings.TrimLeft(dsPath, "/")), "/")
```

Comment:
Why not use path.Split for this?

Reply:
It returns parent directory, we can have several layers of nesting.

Reply:
And you only need to move the first component of dsPath to the remoteName?
That wasn't too clear from the comments.
I guess there can be 
dsName = bucket (what we currently handle)
dsName = bucket/path, which this PR handles by moving the "path" as a prefix to remoteName,

This assumes that a valid bucket name can never contain a '/'. Is that the case?

Reply:
I put additional comment. Valid bucket name should not contain '/', correct.

Reply:
Acording to https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html that is correct.

So makes sense to add some more detail to the comments that we assume that the bucket name does not contain a / and we make the remoteName pick up anything after a /

Reply:
Comment is here https://github.com/lf-edge/eve/pull/2534/files#diff-bdccae192f21fdf85cbdd88b88332042e35ca06096f3844b6cad962e1d575d1bR112

Reply:
But that doesn't state the assumption the code makes that if a Dpath contains a /, then the part before the / is the bucket and the remainder should be viewed as a prefix for the name.

Reply:
Added, please check